felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -4100,6 +4146,25 @@ func (m *bpfEndpointManager) getIfaceTypeFromLink(link netlink.Link) IfaceType {
 	return IfaceTypeData
 }
 
+func (m *bpfEndpointManager) getBondSlaves(masterIfName string) []string {
+	m.ifacesLock.Lock()
+	val, ok := m.nameToIface[masterIfName]
+	m.ifacesLock.Unlock()
+
+	if !ok || val.info.ifaceType != IfaceTypeBond {
+		return []string{}
+	}
+
+	masterIfIndex := val.info.ifIndex
+	slaveDevices := []string{}
+	for name, iface := range m.nameToIface {
```

Comment:
wouldn't it be easier to somehow remember the slaves in the master? When you have a large deploynment with zillions of workload ifaces, you iterate through all of them just to figure out is there are some slaves for this device. You could have the slice of slaves in the master and just return that.